overlord/devicestate/remodel.go

```diff
@@ -137,14 +137,13 @@ func remodelCtx(st *state.State, oldModel, newModel *asserts.Model) (remodelCont
 		// simple context for the simple case
 		remodCtx = &updateRemodelContext{baseRemodelContext{newModel}}
 	case StoreSwitchRemodel:
-		storeSwitchCtx := &newStoreRemodelContext{
-			baseRemodelContext: baseRemodelContext{newModel},
-			st:                 st,
-			deviceMgr:          devMgr,
-		}
-		storeSwitchCtx.store = devMgr.newStore(storeSwitchCtx.deviceBackend())
+		storeSwitchCtx := &newStoreRemodelContext{}
```

Comment:
Silly question - why not something like `storeSwitchCtx := NewDifferentStoreRemodelContext(st, devMgr, newModel)`, i.e. what is the upside of using the delayed "constructor"?

Reply:
Because  reregRemodelContext holds one of these two but not via a pointer. I suppose if we switched to that indirection it might work but otherwise you end up easily confused by things taking pointers to different values.